### PR TITLE
Msys2 mingw better compitability

### PR DESCRIPTION
**Pull request by RyuMaster**

### CODE REVIEW

File: xayagametest/testcase.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from . import premine
 from . import xaya
 
+import tempfile
+import sysconfig
```
Comment (domob1812) [inline xayagametest/testcase.py:14]: These should be sorted alphabetically - please reorder.

File: xayagametest/testcase.py [hunk 2]
```diff
@@ -83,6 +85,14 @@ def main (self):
 
     randomSuffix = "%08x" % random.getrandbits (32)
     self.basedir = os.path.join (self.args.dir, DIR_PREFIX + randomSuffix)
+    
+    if sysconfig.get_platform() == "mingw":
+      self.args.dir = tempfile.gettempdir()
+      self.basedir = os.path.join (self.args.dir, DIR_PREFIX + randomSuffix)
```
Comment (domob1812) [inline xayagametest/testcase.py:91]: I believe the proper way to fix this is updating how the default argument is set in the parser, around here: https://github.com/xaya/libxayagame/blob/74ae5848d1b4620c0d9ea658ba4dc5269f00369b/xayagametest/testcase.py#L58

Probably just removing `DEFAULT_DIR` and instead using `gettempdir()` on every platform is what we should do.

File: xayagametest/testcase.py [hunk 3]
```diff
@@ -83,6 +85,14 @@ def main (self):
...
+      spath = sys.path[0]
+      spath = spath.replace("\\","/")
+      self.args.game_daemon = spath + "/../src/" + self.gameId + "d.exe"
```
Comment (domob1812) [inline xayagametest/testcase.py:94]: This is a hack that is not guaranteed to work with every game.  Instead, the game's test class should pass the right binary as `gameBinaryDefault` already.

File: xayagametest/testcase.py [hunk 4]
```diff
@@ -103,9 +113,18 @@ def main (self):
...
     self.mainLogger.addHandler (logHandler)
     self.mainLogger.addHandler (mainHandler)
+
+    
+    if sysconfig.get_platform() == "mingw":
+      self.mainLogger.info ("msys2 mingw windows build detected, adjustings paths") 
+      self.mainLogger.info ("setting new tempdir as %s" % tempfile.gettempdir()) 
+      self.args.xayad_binary = sysconfig.get_config_var('LIBDIR') + "/daemon/xayad"
+      self.mainLogger.info ("setting new bynary path as %s" % self.args.xayad_binary) 
+      self.mainLogger.info ("setting new game_daemon path as %s" % self.args.game_daemon)
```
Comment (domob1812) [inline xayagametest/testcase.py:123]: Here again, you should fix this where the argument parser default values are set.  Perhaps replace `XAYAD_BINARY_DEFAULT` with a function, and in that function check for the platform.

File: xayagametest/testcase.py [hunk 5]
```diff
@@ -132,13 +151,14 @@ def main (self):
...
                                self.args.xayad_binary)
+                                                          
     self.gamenode = self.createGameNode ()
 
     class RpcHandles:
       xaya = None
       game = None
+      
     self.rpc = RpcHandles ()
-
```
Comment (domob1812) [inline xayagametest/testcase.py:141]: Please revert all these whitespace changes.

File: xayagametest/testcase.py [hunk 6]
```diff
@@ -248,9 +268,17 @@ def createGameNode (self, gameBinary=None):
...
+                    % self.args.game_daemon)    
 
     gameCmd = list (self.runGameWith)
     gameCmd.append (gameBinary)
+    
+    self.log.info ("Tryiung to run gamande in based dir %s"
+                    % self.basedir)    
+
+    self.log.info ("Tryiung to run gamande with params as %s"
+                    % gameCmd)
```
Comment (domob1812) [inline xayagametest/testcase.py:281]: Please revert these debug logs (before we can merge it).

**Review by domob1812 — CHANGES_REQUESTED** [review]
Thanks for the suggestions!  I believe we need to restructure the changes a bit, but in general that seems like a good fix.

**Comment by RyuMaster** [comment]
Got it, will do!

**Pull request closed without merging**